.github/workflows/tests.yml

```diff
@@ -19,17 +19,16 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+      - name: Install Poetry
+        run: pip install poetry
       - name: Setup python
         uses: actions/setup-python@v5
         with:
           python-version: "3.11"
           architecture: x64
-          cache: pip
+          cache: 'poetry'
           cache-dependency-path: "**/pyproject.toml"
```

Comment:
We should remove this line, it's meant to point to the lock file (which is the default)
```suggestion
```